Maestro/src/Microsoft.DotNet.Maestro.WebApi/Controllers/CommitPushedController.cs

```diff
@@ -20,8 +21,27 @@ public class CommitPushedController : ApiController
         private EventRegistrationTable _eventService = new EventRegistrationTable();
         private AzureStorageService _storageService = new AzureStorageService();
 
-        public async Task Post(PushWebHookEvent e)
+        public async Task Post()
         {
+            // The PushWebHookEvent object must be manually deserialized instead of using parameter binding 
+            // because the payload signature verification requires access to the request's raw content.  When
+            // using parameter binding, reading the content in the controller method returns empty.
+            byte[] requestContent = await Request.Content.ReadAsByteArrayAsync();
+
+            try
+            {
+                new RequestValidator(Request, requestContent).VerifyPayloadSignature();
+            }
+            catch (Exception ex)
+            {
+                Trace.TraceError($"An unhandled exception occurred verifying the payload signature : Exception: {ex}");
+
+                throw;
```

Comment:
What handles this exception?  If, for example, the details get returned to the remote caller you could be telling them the correct signature value for their forgery.  If it is just a generic error (or there's no indication) that's fine.